libs/designsystem/src/lib/components/modal/services/modal-outlet.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+import {
+  Router,
+  ActivatedRoute,
+  ChildrenOutletContexts,
+  NavigationEnd,
+  RouterEvent,
+} from '@angular/router';
+import { filter } from 'rxjs/operators';
+
+import { ModalWrapperComponent } from '../modal-wrapper/modal-wrapper.component';
+
+@Injectable({ providedIn: 'root' })
+export class ModalOutlet {
+  resolve$ = this.router.events.pipe(
+    filter((event: RouterEvent) => event instanceof NavigationEnd),
+    filter((event: NavigationEnd) => {
+      return this.isModalRoute(event.url);
+    })
+  );
+
+  constructor(private router: Router, private parentContexts: ChildrenOutletContexts) {}
+
+  destroy() {
+    if (!this.isModalRoute(this.router.url)) return;
+
+    this.router.navigate([
+      this.router.url.split('(')[0], // current route without secondary routes
+      {
+        outlets: {
+          modal: null,
+        },
+      },
+    ]);
+  }
+
+  activate(modalWrapper: ModalWrapperComponent) {
+    if (!this.isModalRoute(this.router.url)) return;
+
+    const routerState = this.getRouterState(this.router);
+    const modalRouterState = this.getModalOutletRouterState(routerState);
+    const outlet = this.parentContexts.getContext('modal').outlet;
+
+    if (!modalRouterState || !outlet) {
+      return;
+    }
+
+    if (outlet.isActivated) {
+      outlet.deactivate();
+    }
+
+    // Set modal title from route configuration
+    modalWrapper.config.title = modalRouterState.snapshot.data.modalTitle;
```

Comment:
Consider supplying only relevant part of the dependency:
```suggestion
    modalConfig.title = modalRouterState.snapshot.data.modalTitle;
```